### PR TITLE
No more smoking in your sleep.

### PR DESCRIPTION
**Pull request by Rivet-the-Zombie**
[Cigarettes are the leading cause of home fire
fatalities in the United States, killing 700 to
900 people per year.](http://www.pfa.org/firesafe/CFSC_facts.pdf)

This adds a check to active cigarette handling which causes the
character to drop their smoke if they fall asleep while consuming it.

### CODE REVIEW

**Comment by BorkBorkGoesTheCode** [comment]
Crispy has fixed astyle #16928

File: src/item.cpp [hunk 1]
```diff
@@ -5421,6 +5422,13 @@ bool item::process_litcig( player *carrier, const tripoint &pos )
             g->m.add_item_or_charges( tripoint( pos.x + rng( -1, 1 ), pos.y + rng( -1, 1 ), pos.z ), *this, 2 );
             return true; // removes the item that has just been added to the map
         }
+
+        if(  carrier->has_effect( effect_sleep ) ) {
+            carrier->add_msg_if_player( m_bad, _( "You fall asleep and drop your %s." ),
+                                        tname().c_str() );
+            g->m.add_item_or_charges( tripoint( pos.x + rng( -1, 1 ), pos.y + rng( -1, 1 ), pos.z ), *this, 2 );
```
Comment (illi-kun) [inline src/item.cpp]: Just a guess about potential issue: what happens when this tripoint refers on tile which cannot contain any items?

File: src/item.cpp [hunk 2]
```diff
@@ -5421,6 +5422,13 @@ bool item::process_litcig( player *carrier, const tripoint &pos )
             g->m.add_item_or_charges( tripoint( pos.x + rng( -1, 1 ), pos.y + rng( -1, 1 ), pos.z ), *this, 2 );
             return true; // removes the item that has just been added to the map
         }
+
+        if(  carrier->has_effect( effect_sleep ) ) {
```
Comment (illi-kun) [inline src/item.cpp]: You have an excessive whitespace here (right after ```if(```).

File: src/item.cpp (hunk 1 above)
Comment (Rivet-the-Zombie) [inline src/item.cpp]: Presumably the same thing that happens when the shakes trigger it above.

**Comment by Mecares** [comment]
Is the dropped cigarette actually able to start a fire?

**Comment by Rivet-the-Zombie** [comment]
It has a chance of doing so if it falls on something sufficiently flammable.

**Comment by Coolthulhu** [comment]
Currently that's a rather broad category - pretty sure a blanket/bed/sofa will all qualify.

**Comment by kevingranade** [comment]
> broad category

As they should? A reasonable rate would be nice, but having a burning
object falling on you when sleeping is pretty intrinsically dangerous.
Note, not "instant immolation" dangerous, but frankly that's the source of
the similar ,"drop match on sock to roast zombie" bug.

What's good for the goose and all that.

**Comment by Rivet-the-Zombie** [comment]
Test characters wake up with singed feet, not full-body burns or anything.

**Pull request merged**